### PR TITLE
Fix #5506 NullPointerException in StudyOptionsFragment

### LINKED ISSUE #5506: [2.9 ACRA] NullPointerException in StudioOptionsFragment.initAllContentViews
https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/8392a87c-9f5a-4243-840d-7a921dfe3577

```
java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.View android.view.View.findViewById(int)' on a null object reference
at com.ichi2.anki.StudyOptionsFragment.initAllContentViews(StudyOptionsFragment.java:3)
at com.ichi2.anki.StudyOptionsFragment.access$400(StudyOptionsFragment.java:1)
at com.ichi2.anki.StudyOptionsFragment$3.onPostExecute(StudyOptionsFragment.java:11)
at com.ichi2.async.DeckTask$TaskListener.onPostExecute(DeckTask.java:1)
at com.ichi2.async.DeckTask.onPostExecute(DeckTask.java:3)
at com.ichi2.async.DeckTask.onPostExecute(DeckTask.java:1)
at android.os.AsyncTask.finish(AsyncTask.java:695)
at android.os.AsyncTask.access$600(AsyncTask.java:180)
at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:712)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:214)
at android.app.ActivityThread.main(ActivityThread.java:7080)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
```

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description

There was a NPE assuming that the view was not null. This assumption does not hold.

We also remove an unused method (loading preferences) on View Load. This was removed in #555 (https://github.com/ankidroid/Anki-Android/commit/2da08a086c417acc42de813d2714a3ce7261947f).

## Fixes
Fixes #5506

## Approach

* Check for null and exit before we perform any view logic.
* Mark variable as nullable so it won't happen again

## How Has This Been Tested?

This is not tested, since I could not replicate the failure condition on my device.

Please see each commit for the logical steps in refactoring.

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.java [hunk 1]
```diff
@@ -487,13 +485,6 @@ private void dismissProgressDialog() {
         }
     }
 
-
-    public SharedPreferences restorePreferences() {
-        SharedPreferences preferences = AnkiDroidApp.getSharedPrefs(getActivity().getBaseContext());
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.java]: So strange - I checked that method to see if there were any side effects (that we wanted) to explain why we get a prefs object and then ignore it everywhere, and nothing. I love seeing dead code go away, thanks

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.java]: To any other readers: Usage removed in 2da08a0, but method remained.

File: AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.java [hunk 2]
```diff
@@ -189,6 +190,7 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
             return null;
         }
         mStudyOptionsView = inflater.inflate(R.layout.studyoptions_fragment, container, false);
+        assert mStudyOptionsView != null;
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.java]: assert doesn't work in the android JVM so this is something we don't use in AnkiDroid
I believe the Android Studio 4.x toolchain (including gradle plugins) which actually start making assert work via the extra de-sugaring they are implementing but that's still in alpha
Either way, if we want assert behavior we Timber.e and throw RuntimeException, it's effectively the same

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.java]: Thanks for the heads up

FYI: We use assert in libAnki. That might need an issue.

I was using an assert to fix code quality warnings over use of a `@Nullable`. Will fix.

**Comment by mikehardy** [comment]
I think we're all caught up, and we're definitely at the maximum velocity of change for a little bit. I'm going to kick out a new alpha, pick existing alpha stuff to a new 2.9.6beta1 and after the new alpha simmers do the same for another beta while doing translations - so the next few days will be slower except for any needed fixes on recent merges to make sure things don't fly apart

**Pull request merged**